scopes/dependencies/dependency-resolver/manifest/deduping/hoist-dependencies.ts

```diff
@@ -429,6 +429,9 @@ function groupByRangeOrVersion(indexItems: PackageNameIndexComponentItem[]): Ite
       throw new Error(`fatal: the version "${item.range}" originated from a dependent "${item.origin}" is invalid semver range.
 this is a temporary issue with unsupported snaps (hashes) on the registry and will be fixed very soon`);
     }
+    // parseRange does not support `*` as version
+    // `*` does not affect resulted version, it might be just ignored
+    if (validRange === '*') return;
```

Comment:
you might need to `result.ranges.push(item);` before you return
because if there is only one dependency it might break, but it needs to be tested. it might work and can be ignored.